include/deal.II/lac/constraint_matrix.templates.h

```diff
@@ -670,7 +683,7 @@ distribute_local_to_global (const Vector<LocalType>      &local_vector,
                 continue;
 
               const ConstraintLine &position_j =
-                lines[lines_cache[calculate_line_index(local_dof_indices[j])]];
+                lines[lines_cache[calculate_line_index(local_dof_indices_row[j])]];
 
               for (size_type q=0; q<position_j.entries.size(); ++q)
                 {
```

Comment:
The code that follows below should be handled correctly again because you got the correct local_dof_indices_row.